test/e2e/operator_test.go

```diff
@@ -455,118 +280,76 @@ u3YLAbyW/lHhOCiZu2iAI8AbmXem9lW6Tr7p/97s0w==
 //
 // See also: https://github.com/kubernetes/kubernetes/pull/75210
 func TestIngressControllerScale(t *testing.T) {
-	cl, ns, err := getClient()
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	name := types.NamespacedName{Namespace: ns, Name: manifests.DefaultIngressControllerName}
-	err = waitForIngressControllerCondition(cl, 15*time.Second, name, defaultAvailableConditions...)
-	if err != nil {
-		t.Errorf("failed to observe expected conditions: %v", err)
-	}
-
 	ic := &operatorv1.IngressController{}
-	if err := cl.Get(context.TODO(), name, ic); err != nil {
-		t.Fatalf("failed to get ingresscontroller %s: %v", name, err)
+	if err := kclient.Get(context.TODO(), defaultName, ic); err != nil {
+		t.Fatalf("failed to get default ingresscontroller: %v", err)
+	}
+	if err := waitForIngressControllerCondition(kclient, 5*time.Minute, defaultName, defaultAvailableConditions...); err != nil {
+		t.Fatalf("failed to observe expected conditions: %v", err)
 	}
 
-	deploymentName := controller.RouterDeploymentName(ic)
+	// TODO: desired replicas isn't actually persisted in our API...
```

Comment:
Is this comment referring to the same issue as the comment at the top of the function?

Reply:
No, it's referring to the fact that there is literally nowhere on the ingresscontroller resource to see what the effective "desired" replica count...